pkg/provider/kubernetes/ingress/client.go

```diff
@@ -275,8 +297,53 @@ func extensionsToNetworking(ing marshaler) (*networkingv1beta1.Ingress, error) {
 	return ni, nil
 }
 
+func addServiceFromV1Beta1(ing *networkingv1.Ingress, old networkingv1beta1.Ingress) {
+	if old.Spec.Backend != nil {
+		port := networkingv1.ServiceBackendPort{}
+		if old.Spec.Backend.ServicePort.Type == intstr.Int {
+			port.Number = old.Spec.Backend.ServicePort.IntVal
+		} else {
+			port.Name = old.Spec.Backend.ServicePort.StrVal
+		}
+
+		ing.Spec.DefaultBackend = &networkingv1.IngressBackend{
+			Service: &networkingv1.IngressServiceBackend{
+				Name: old.Spec.Backend.ServiceName,
```

Comment:
Service name is also optional, we may not have to create the IngressBackend when it's an empty string

Reply:
done :)